debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlTaskContext.java

```diff
@@ -195,6 +198,20 @@ public boolean useMinimalSnapshotLocking() {
         return config.getBoolean(MySqlConnectorConfig.SNAPSHOT_MINIMAL_LOCKING);
     }
 
+    public Optional<String> getSnapshotSelectOverride(String tableId) {
```

Comment:
How about encapsulating the entire select statement determination in this method? It'd either return the default or the override if present.